mono/mini/intrinsics.c

```diff
@@ -755,6 +756,13 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 			EMIT_NEW_BIALU_IMM (cfg, ins, OP_PADD_IMM, dreg, args [0]->dreg, MONO_STRUCT_OFFSET (MonoArray, vector));
 			return ins;
 		}
+		else if (!strcmp (cmethod->name, "GetElementSize")) {
+			int dreg = alloc_ireg (cfg);
+			int vt_reg = alloc_preg (cfg);
+			MONO_EMIT_NEW_LOAD_MEMBASE_FAULT (cfg, vt_reg, args [0]->dreg, MONO_STRUCT_OFFSET (MonoObject, vtable));
+			EMIT_NEW_LOAD_MEMBASE (cfg, ins, OP_LOAD_MEMBASE, dreg, vt_reg, MONO_STRUCT_OFFSET (MonoVTable, klass));
+			return mono_emit_jit_icall (cfg, mono_array_element_size, &ins);
```

Comment:
can't you just access size directly without the icall?

Reply:
That was my first thought but it may not be initialized. Not sure if there's an easy solution for that.

Reply:
Well, you can emit a condition then 🙂, and emit the icall if `m_class_is_size_inited` is 0 as a slow path.

Reply:
It should be initialized already if there are instances of the array.

Reply:
I'll rework it.

Reply:
The icall had to initialize it because it was working on Type objects.